pkg/stanza/fileconsumer/finder_test.go

```diff
@@ -123,6 +124,206 @@ func TestFinder(t *testing.T) {
+								RegexKey:  "value",
+								Ascending: false,
+							},
+						},
+					},
+				},
+			},
+			expected: []string{"err.123456789.log"},
+		},
+		{
+			name:    "Numeric Sorting Ascending",
+			files:   []string{"err.123456789.log", "err.123456788.log", "err.123456787.log", "err.123456786.log"},
+			include: []string{"err.*.log"},
+			exclude: []string{},
+			filterSortRule: OrderingCriteria{
+				Regex: `err\.(?P<value>\d+).*log`,
+				SortBy: []SortRuleImpl{
+					{
+						NumericSortRule{
+							BaseSortRule: BaseSortRule{
+								RegexKey:  "value",
+								Ascending: true,
+							},
+						},
+					},
+				},
+			},
+			expected: []string{"err.123456786.log"},
+		},
+		{
+			name:    "Alphabetical Sorting",
+			files:   []string{"err.a.log", "err.b.log", "err.c.log", "err.d.log"},
+			include: []string{"err.*.log"},
+			exclude: []string{},
+			filterSortRule: OrderingCriteria{
+				Regex: `err\.(?P<value>[a-zA-Z]+).*log`,
+				SortBy: []SortRuleImpl{
+					{
+						AlphabeticalSortRule{
+							BaseSortRule: BaseSortRule{
+								RegexKey:  "value",
+								Ascending: false,
+							},
+						},
+					},
+				},
+			},
+			expected: []string{"err.d.log"},
+		},
+		{
+			name:    "Alphabetical Sorting Ascending",
+			files:   []string{"err.a.log", "err.b.log", "err.c.log", "err.d.log"},
+			include: []string{"err.*.log"},
+			exclude: []string{},
+			filterSortRule: OrderingCriteria{
+				Regex: `err\.(?P<value>[a-zA-Z]+).*log`,
+				SortBy: []SortRuleImpl{
+					{
+						AlphabeticalSortRule{
+							BaseSortRule: BaseSortRule{
+								RegexKey:  "value",
+								Ascending: true,
+							},
+						},
+					},
+				},
+			},
+			expected: []string{"err.a.log"},
+		},
+		{
+			name:    "Multiple Sorting - timestamp priority sort",
+			files:   []string{"err.a.123456786.2023020610.log", "err.b.123456789.2023020609.log", "err.c.123456788.2023020612.log", "err.d.123456787.2023020611.log"},
+			include: []string{"err.*.log"},
+			exclude: []string{},
+			filterSortRule: OrderingCriteria{
+				Regex: `err\.(?P<alpha>[a-zA-Z])\.(?P<number>\d+)\.(?P<time>\d{10})\.log`,
+				SortBy: []SortRuleImpl{
+					{
+						AlphabeticalSortRule{
+							BaseSortRule: BaseSortRule{
+								RegexKey:  "alpha",
+								Ascending: false,
+							},
+						},
+					},
+					{
+						NumericSortRule{
+							BaseSortRule: BaseSortRule{
+								RegexKey:  "number",
+								Ascending: false,
+							},
+						},
+					},
+					{
+						TimestampSortRule{
+							BaseSortRule: BaseSortRule{
+								RegexKey:  "time",
+								Ascending: false,
+							},
+							Location: "UTC",
+							Format:   `%Y%m%d%H`,
+						},
+					},
+				},
+			},
+			expected: []string{"err.c.123456788.2023020612.log"},
+		},
+		{
+			name:    "Multiple Sorting - alpha priority sort",
+			files:   []string{"err.a.123456786.2023020610.log", "err.b.123456789.2023020609.log", "err.c.123456788.2023020612.log", "err.d.123456787.2023020611.log"},
```

Comment:
I think we need more files here in order to be sure the sorts are being applied in the correct order. Looking at this as binary (`a|b`, `1|2`, `2023..01|2023..02`), we can make each combination and list them in an order where we only get the expected result if applying the sorts correctly:

```
"err.b.1.2023020601.log"
"err.b.2.2023020601.log"
"err.a.1.2023020601.log"
"err.a.2.2023020601.log"
"err.b.1.2023020602.log"
"err.a.2.2023020602.log"
"err.b.2.2023020602.log" // expected
"err.a.1.2023020602.log"
```

sorted by `1|2` descending:

```
"err.b.2.2023020601.log"
"err.a.2.2023020601.log"
"err.a.2.2023020602.log"
"err.b.2.2023020602.log" // expected
"err.b.1.2023020601.log"
"err.a.1.2023020601.log"
"err.b.1.2023020602.log"
"err.a.1.2023020602.log"
```

then `2023..01|2023..02` descending:

```
"err.a.2.2023020602.log"
"err.b.2.2023020602.log" // expected
"err.b.1.2023020602.log"
"err.a.1.2023020602.log"
"err.b.2.2023020601.log"
"err.a.2.2023020601.log"
"err.b.1.2023020601.log"
"err.a.1.2023020601.log"
```

finally `a|b` ascending:

```
"err.b.2.2023020602.log" // expected
"err.b.1.2023020602.log"
"err.b.2.2023020601.log"
"err.b.1.2023020601.log"
"err.a.2.2023020602.log"
"err.a.1.2023020602.log"
"err.a.2.2023020601.log"
"err.a.1.2023020601.log"
```

Reply:
I've altered these tests to try to capture what you're going for.

Reply:
Ah, I see I applied the last sort incorrectly and so the expected result was wrong. Fixing it for posterity but yours looks good.